src/Acme.UI/Models/Authentication/MemberViewModel.cs

```diff
@@ -0,0 +1,201 @@
+
+        public string Address
+        {
+            get
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == addressAlias);
+                if (found != null)
+                {
+                    return found.Value;
+                }
+                return null; 
+            }
+
+            set
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == addressAlias);
+                if (found != null)
+                {
+                    RegisterModel.MemberProperties.Remove(found);
+                }
+                RegisterModel.MemberProperties.Add(new UmbracoProperty { Alias = addressAlias, Value = value});
+            }
+        }
+
+        public string Phone
+        {
+            get
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == phoneAlias);
+                if (found != null)
+                {
+                    return found.Value;
+                }
+                return null;
+            }
+
+            set
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == phoneAlias);
+                if (found != null)
+                {
+                    RegisterModel.MemberProperties.Remove(found);
+                }
+                RegisterModel.MemberProperties.Add(new UmbracoProperty { Alias = phoneAlias, Value = value });
+            }
+        }
+
+        public bool IsApproved
+        {
+            get
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == umbracoIsApprovedPropAlias);
+                if (found != null)
+                {
+                    return found.Value != "0";
+                }
+                return true; // it is true in all other cases while registering the user
+            }
+
+            set
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == umbracoIsApprovedPropAlias);
+                if (found != null)
+                {
+                    RegisterModel.MemberProperties.Remove(found);
+                }
+                RegisterModel.MemberProperties.Add(new UmbracoProperty { Alias = umbracoIsApprovedPropAlias, Value = (value ? "1" : "0") });
+            }
+        }
+
+        public string ConfirmPassword { get; set; }
+
+        public string DisplayNameTitle { get; set; }
+
+        public string UserNameTitle { get; set; }
+
+        public string EmailTitle { get; set; }
+
+        public string AddressTitle { get; set; }
+
+        public string PhoneTitle { get; set; }
+
+        public string PasswordTitle { get; set; }
+
+        public string ConfirmPasswordTitle { get; set; }
+
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            if (string.IsNullOrWhiteSpace(DisplayName))
```

Comment:
For fun can you please come up with a way to not repeat yourself with the if statements and yields for every single field?
If you dont know how lets have a discussion and brainstorm ideas.